hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1602,15 +1602,15 @@ func (r *HostedClusterReconciler) reconcileCAPIManager(ctx context.Context, crea
 
 	// Reconcile CAPI manager deployment
 	capiImage, err := hyperutil.GetPayloadImage(ctx, hcluster, ImageStreamCAPI, pullSecretBytes)
-	if err != nil {
-		return fmt.Errorf("failed to retrieve capi image: %w", err)
-	}
 	if envImage := os.Getenv(images.CAPIEnvVar); len(envImage) > 0 {
 		capiImage = envImage
 	}
 	if _, ok := hcluster.Annotations[hyperv1.ClusterAPIManagerImage]; ok {
 		capiImage = hcluster.Annotations[hyperv1.ClusterAPIManagerImage]
 	}
+	if capiImage == "" && err != nil {
```

Comment:
It's the order in `capiImage == "" && err != nil` that is confusing, `err != nil && capiImage == ""` would be more readable.

Reply:
That order will not work; we want it to short circuit on the `capiImage == ""` first. 

With `err != nil && capiImage == ""`, if it fails to get the `capiImage` from the release image but gets it from the environment variable or hc annotation, it will return the error rather than continuing on.